### PR TITLE
Proportional, not fixed, LUC losses

### PR DESCRIPTION
**Pull request by bpbond**
In all versions of Hector to date, land use changes (emissions or uptake) cause carbon to leave (or enter) the terrestrial pools in fixed and user-definable fractions: `f_lucv`, `f_lucd` and implicitly `f_lucs` for the fraction from/to vegetation, detritus, and soil respectively.

However in #638 Dawn made the case that this is both _ad hoc_ and uncertain, depending heavily on the exact disturbance type; and that it made more sense — was simpler and more conservative — to simply pull/add C proportionately to the pool sizes. This also has some nice properties in terms of the model's stability after a LUC pulse — see #646 #643 .

This PR makes this change, shifting the land model to use proportional LUC changes — from #639 — but now split into their own PR for easier evaluation.

** Note that this is branched from #646 **, as I assume we will definitely merge that.

Changes:

diff

Detail:

diff-detail

### CODE REVIEW

**Comment by bpbond** [comment]
This PR against the updated `v3_dev`:

diff

Detail:

diff_detail

**Comment by bpbond** [comment]
Oh interesting, look at the `HL_pH` panel above.

**Comment by kdorheim** [comment]
@bpbond could you please explain a bit more what you mean by "Proportional LUC" I want to make sure that I fully understand what is going on here and that our future selves will know what was meant by "Proportional LUC"

**Comment by bpbond** [comment]
@kdorheim Updated the primary comment above; is that more complete and clearer?

**Review by kdorheim — COMMENTED** [review]
I have some questions can we go over them tomorrow during our meeting?

File: src/simpleNbox-runtime.cpp [hunk 1]
```diff
@@ -304,22 +300,27 @@ void SimpleNbox::stashCValues(double t, const double c[]) {
...
+    // Calculate luc emissons
+    const double veg_frac = veg_c[biome].value(U_PGC) / total;
+    const double det_frac = detritus_c[biome].value(U_PGC) / total;
+    const double soil_frac = soil_c[biome].value(U_PGC) / total;
+    // Note we don't need to include 'wt' here because the veg_frac, det_frac,
+    // and soil_frac fractions calculated above handle that
+    fluxpool luc_fva_biome_flux = yf * veg_c[biome].flux_from_fluxpool(luc_e_untracked * veg_frac);
+    fluxpool luc_fda_biome_flux = yf * detritus_c[biome].flux_from_fluxpool(luc_e_untracked * det_frac);
+    fluxpool luc_fsa_biome_flux = yf * soil_c[biome].flux_from_fluxpool(luc_e_untracked * soil_frac);
+    // Calculate luc uptake
```
[the remark this answers is not in the document]
Comment (ssmithClimate) [reply src/simpleNbox-runtime.cpp:319]: Reforestation would likely increase the detritus pool, which is smaller in other land use types. It depends on the type of forest. An oak forest will have much more in this pool as oak leaves have (I believe) higher lignin content and will accumulate, while other forest types might not have as much.

Conversion from intensely tilled cropland to native grassland might significantly increase the soil carbon pool, albeit over a quite long time frame (depending on the tillage methods previously used for the cropland).

[the remark this answers is not in the document]
Comment (bpbond) [reply src/simpleNbox-runtime.cpp]: Yes, but @kdorheim 's point is that those detritus and soil C increases have to flow through the vegetation _first_.

File: src/simpleNbox-runtime.cpp [hunk 2]
```diff
@@ -304,22 +300,27 @@ void SimpleNbox::stashCValues(double t, const double c[]) {
...
-        soil_c[biome].flux_from_fluxpool((luc_e_untracked * f_lucs) * wt);
-    atmos_c = atmos_c + luc_fva_biome_flux - luc_fav_flux * wt;
-    atmos_c = atmos_c + luc_fda_biome_flux - luc_fad_flux * wt;
-    atmos_c = atmos_c + luc_fsa_biome_flux - luc_fas_flux * wt;
 
+    // Calculate luc emissons
+    const double veg_frac = veg_c[biome].value(U_PGC) / total;
+    const double det_frac = detritus_c[biome].value(U_PGC) / total;
+    const double soil_frac = soil_c[biome].value(U_PGC) / total;
+    // Note we don't need to include 'wt' here because the veg_frac, det_frac,
```
[the remark this answers is not in the document]
Comment (bpbond) [reply src/simpleNbox-runtime.cpp:315]: Clarified this in 4dc25f77 - thanks!

File: src/simpleNbox-runtime.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bpbond) [reply src/simpleNbox-runtime.cpp]: P.S thanks @ssmithClimate !

[the remark this answers is not in the document]
Comment (ssmithClimate) [reply src/simpleNbox-runtime.cpp]: @bpbond yes indeed!  Just noting that changes in parameters (e.g. timescales) also play a role. Cropland might sometimes be more productive (more C flow) than native grassland, but if you till it every year you won't get the carbon accumulation.

**Comment by bpbond** [comment]
Updated change-from-v3_dev plot:

diff

Detail:

diff_detail

**Comment by bpbond** [comment]
Over to you @kdorheim for assessment and, if things look good, approval and merging.

**Review by kdorheim — APPROVED** [review]

**Pull request merged**